options.go

```diff
@@ -426,8 +427,12 @@ func (o *GrepOptions) Validate(w *Worktree) error {
 }
 
 // PlainOpenOptions describes how opening a plain repository should be
+
 // performed.
 type PlainOpenOptions struct {
+	// Storage layer options.
```

Comment:
Maybe we should keep this out of `Plain*` methods and avoid adding `Close` to general interface. Just adding it to the `filesystem` implementation would be enough, and this option would be available only by creating the `filesystem` storage directly.

This way we avoid redundancy in other implementatons or end up exposing a lot of storage-specific options here.

Reply:
Done